asf_welsh_energy_consultation/pipeline/process_data.py

```diff
@@ -0,0 +1,256 @@
+# File: asf_welsh_energy_consultation/pipeline/process_data.py
+"""
+Functions to process and augment data.
+"""
+
+import pandas as pd
+
+from asf_welsh_energy_consultation.getters.get_data import *
+
+
+# PROCESSING MCS
+
+
+def get_enhanced_mcs():
+    """Get dataset of domestic MCS installations with attached off-gas, country and rurality fields.
+
+    Returns:
+        pd.DataFrame: Dataset as described above.
+    """
+    mcs = get_mcs_domestic()
+    og = get_offgas()
+    countries = get_countries()
+    rural = get_rurality()
+
+    # join with off-gas data
+    mcs = mcs.merge(og, on="postcode", how="left")
+    mcs["off_gas"] = mcs["off_gas"].fillna("On gas").replace({True: "Off gas"})
+
+    # join with regions in order to filter to Wales
+    mcs = mcs.merge(countries, on="postcode", how="left")
+    mcs = mcs.loc[mcs["country"] == "Wales"].reset_index(drop=True)
+    # 1203 records with no match - 273 are Northern Ireland which leaves 918
+    # Some will be new postcodes (new build developments)
+    # and some may be expired postcodes
+    # In future, implement new solution that uses outward codes
+
+    # join with rurality data
+    mcs = mcs.merge(rural, on="postcode", how="left")
+    # only 13 postcodes lost in this merge
+
+    # add custom rurality column
+    mcs["rurality_7"] = mcs["rurality_10_label"].replace(
+        {
+            "Urban city and town": "Urban",
+            "Urban major conurbation": "Urban",
+            "Urban city and town in a sparse setting": "Urban",
+            "Urban minor conurbation": "Urban",
+        }
+    )
+
+    return mcs
+
+
+# load enhanced MCS as part of this script, so only needs to be done once
+mcs = get_enhanced_mcs()
+
+
+def cumsums_by_variable(variable, new_var_name, data=mcs):
```

Comment:
```suggestion
def cumsums_by_variable(variable, new_var_name, data=get_enhanced_mcs()):
```
just a suggestion - not sure this is even good practice tbh, but just suggesting this because a bit confusing to have `mcs` defined in line 55, because we have different mcs datasets. Or maybe, alternatively, we could call the variable `enhanced_mcs` in line 55 and pass the default here as `enhanced_mcs`?

Reply:
I'll clarify this by calling it `enhanced_mcs`